lms/static/js/learner_dashboard/views/course_enroll_view.js

```diff
@@ -20,23 +20,105 @@
                 tpl: HtmlUtils.template(pageTpl),
 
                 events: {
-                    'click .enroll-button': 'handleEnroll'
+                    'click .enroll-button': 'handleEnroll',
+                    'change .run-select': 'handleRunSelect',
                 },
 
                 initialize: function(options) {
-                    if (options.$el){
-                        this.$el = options.$el;
-                        this.render();
+                    this.$parentEl = options.$parentEl;
+                    this.enrollModel = options.enrollModel;
+                    this.urlModel = options.urlModel;
+                    this.render();
+                    if(this.urlModel){
+                        this.trackSelectionUrl = this.urlModel.get('track_selection_url');
+                        this.dashboardUrl = this.urlModel.get('dashboard_url');
+                        this.verificationUrl = this.urlModel.get('id_verification_url');
                     }
                 },
 
+                updateIsEnrolled: function(){
+                    
+                },
+
                 render: function() {
-                    var filledTemplate = this.tpl(this.model.toJSON());
-                    HtmlUtils.setHtml(this.$el, filledTemplate);
+                    var filledTemplate;
+                    if (this.$parentEl &&
+                        this.enrollModel &&
+                        this.model.get('course_key')){
+
+                        filledTemplate = this.tpl(this.model.toJSON());
+                        HtmlUtils.setHtml(this.$el, filledTemplate);
+                        HtmlUtils.setHtml(this.$parentEl, HtmlUtils.HTML(this.$el));
+                        this.$('.run-select').val(this.model.get('run_key'));
```

Comment:
This should be done in the template, if the option === the run_key set it as selected.